### PR TITLE
Fix infinite execution of partition selector

### PR DESCRIPTION
**Pull request by dhanashreek89**
This fixes a bug introduced in https://github.com/greenplum-db/gpdb/commit/e378d84bf32922b324a8ce6e8a66be358a0b805f where the partition selector may go in an infinite loop because it never got a chance to actually _select_ the partitions.

This fixes https://github.com/greenplum-db/gpdb/issues/2249

### CODE REVIEW

**Comment by dhanashreek89** [comment]
@karthijrk @foyzur @hsyuan @hardikar @vraghavan78 @d @bhuvnesh2703 @oarap  @hlinnaka @danielgustafsson 
Please review.

File: src/backend/executor/nodePartitionSelector.c [hunk 1]
```diff
@@ -195,7 +195,10 @@ ExecPartitionSelector(PartitionSelectorState *node)
 	econtext->ecxt_outertuple = inputSlot;
 	econtext->ecxt_scantuple = inputSlot;
 
-	candidateOutputSlot = ExecProject(node->ps.ps_ProjInfo, NULL);
+	if(NULL != inputSlot)
```
Comment (hsyuan) [inline src/backend/executor/nodePartitionSelector.c]: 0. space after `if`
0. `NULL` on the right side to abide by the postgres way.

File: src/test/regress/sql/bfv_partition.sql [hunk 2]
```diff
@@ -794,6 +794,25 @@ EVERY ('2 mons'::interval)
 explain analyze select a.* from mpp8031 a, mpp8031 b where a.oid = b.oid;
 drop table mpp8031;
 
+-- Test Query on Partition table when optimizer_static_partition_selection is OFF.
+-- start_ignore
+DROP TABLE IF EXISTS partitioned_table;
+-- end_ignore
+
+CREATE TABLE partitioned_table (a int)
```
Comment (hsyuan) [inline src/test/regress/sql/bfv_partition.sql]: You don't need to create new table. Just reuse table bar on L731.

Comment (dhanashreek89) [inline src/test/regress/sql/bfv_partition.sql]: Table bar is getting dropped later on L774.

**Comment by dhanashreek89** [comment]
Thanks @hsyuan. Addressed the comments.

File: src/test/regress/sql/bfv_partition.sql (hunk 2 above)
Comment (hsyuan) [inline src/test/regress/sql/bfv_partition.sql]: hahaha, you can put your test before L774. But that's ok. It is not required. :)

File: src/backend/executor/nodePartitionSelector.c [hunk 3]
```diff
@@ -195,7 +195,10 @@ ExecPartitionSelector(PartitionSelectorState *node)
 	econtext->ecxt_outertuple = inputSlot;
 	econtext->ecxt_scantuple = inputSlot;
```
Comment (ppggff) [inline src/backend/executor/nodePartitionSelector.c]: why leaving 'ResetExprContext and set given input tuple' out of 'if (NULL != inputSlot)' below?

Comment (ppggff) [inline src/backend/executor/nodePartitionSelector.c]: @dhanashreek89

**Pull request merged**